### DISCUSSION TITLE
Introduce docs versioning of some sort

### DISCUSSION THREAD

**Discussion by ItamarShDev** [remix-run/remix#2563] [Proposals]
Case study
---
I have started to work on a small playground project by mimicking my Next.js based blog with Remix.run too.    
This helps me better understand the library.

At the time of starting , the Remix amazing docs had a guide on creating a blog with .md files, which was exactly what i have needed!

I have started  to work on it on some nights and then have contracted COVID-19 and left the computer for a week.

When i was well enough to come back, i have noticed that the [blog guide](https://remix.run/docs/en/v1/tutorials/blog) is different.    
I was able to find the original one in the repo easily [here](https://github.com/remix-run/remix/blob/82432f72e2270267ca932d885a92fe257478f41e/docs/tutorials/blog.md), and continue working.

This gave me an idea:
### Allow users to see old versions of the docs

How
---
I have compiled 3 ideas, but more may be available

## 1. Route as source of truth
Use the url to allow to go back to old versions.    
Either by using the `v1` in the url, or with any other url params.

## 2. Archive
Add Archive section that has at least the last major change.

## 3. Keep old versions 
In the side bar, allow listing different versions 

<img width="376" alt="image" src="https://user-images.githubusercontent.com/953371/161017561-8cd7cc45-f6b6-479b-9a0c-a16b72257030.png">
 instead of 
<img width="371" alt="image" src="https://user-images.githubusercontent.com/953371/161015798-b441f50d-7c4e-43b9-9b94-4693dfe87a2d.png">

**Comment by gerbyzation** [comment]
The pain of this is quite noticable with nextjs for example, they are changing more APIs and the website only has the latest version available so if you're still on v12/13 you might have to dive into source to find documention. It would be really good if you could switch to previous major versions of the documentation

**Comment by sergiodxa** [comment]
The docs already support versioning, but the version picker only list the last one of each major, however you can change the URL to get any version, eg this is v1.1.1 docs https://remix.run/docs/en/v1.1.1.

**Reply by gerbyzation** [reply]
Right, I completely glanced over that as a version selector, as it says "main" while I was looking for a version indicator 😄
